File: su/2026-05/15_pgr_주식가격.py
def solution(prices):
    n = len(prices)
    answer = [0] * n
    stack = []
    
    for i in range(n):
        if not stack:
            stack.append((prices[i], i))
            continue
            
            
        while stack and stack[-1][0] > prices[i]:  # 가격이 떨어진 경우
            p, idx = stack.pop()
            answer[idx] = i - idx
            
        stack.append((prices[i], i))

            
    while stack:  # 끝까지 가격이 안 떨어진 경우
        p, idx = stack.pop()
        answer[idx] = n - 1 - idx
    
    return answer

File: su/2026-05/15_pgr_주식가격.java
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        int n = prices.length;
        int[] answer = new int [n];
        
        ArrayDeque<int[]> stack = new ArrayDeque<>();  // ❌ 타입 없으면 꺼낼 때 Object로 나옴 → [1] 인덱싱 불가

        
        for (int i = 0; i < n; i++) {
            while (!stack.isEmpty() && stack.peekLast()[1] > prices[i]) {
                int[] newInfo = stack.pollLast();
                int idx = newInfo[0];
                answer[idx] = i - idx;
            }
            
            stack.add(new int[] {i, prices[i]});
        }
        
        while (!stack.isEmpty()) {
            int[] newInfo = stack.pollLast();
            int idx = newInfo[0];
            answer[idx] = n - idx - 1;
        }
        
        return answer;
    }
}